build.gradle

```diff
@@ -1,14 +1,20 @@
+buildscript {
+    dependencies {
+        classpath "gradle.plugin.com.github.spotbugs.snom:spotbugs-gradle-plugin:4.0.0"
+    }
+}
+
 plugins {
     id "com.jfrog.artifactory" version "3.1.1"
+    id "java-library"
```

Comment:
might as well convert all the `apply plugin` to use `plugins {}` while we're at it

Reply:
Sadly I think last I looked at this it didn't work for all plugins - a few older ones or legacy style ones aren't compatible. Something-something need the old school `buildscript` dependency thing to work or so. I tried it _real_ quick - and it does seem like the spotbugs plugin works the new way now. But the config / codeMetrics stuff isn't ready so I couldn't fully confirm